core/io/http_client.cpp

```diff
@@ -96,6 +96,11 @@ Error HTTPClient::connect_to_host(const String &p_host, int p_port, bool p_ssl,
 void HTTPClient::set_connection(const Ref<StreamPeer> &p_connection) {
 	ERR_FAIL_COND_MSG(p_connection.is_null(), "Connection is not a reference to a valid StreamPeer object.");
 
+	if (ssl) {
+		ERR_FAIL_NULL_MSG(Object::cast_to<StreamPeerSSL>(p_connection.ptr()),
+				"Connection is not a reference to a valid StreamPeerSSL object.");
+	}
```

Comment:
Thinking about it, it's better to do:
```suggestion
	ERR_FAIL_COND_MSG(ssl && Object::cast_to<StreamPeerSSL>(p_connection.ptr() == nullptr),
			"Connection is not a reference to a valid StreamPeerSSL object.");
```
`ERR_FAIL_*` uses `unlikely()` for the condition so it's better optimized, and the "C++ error" in the debugger will then include both conditions for failure.

Reply:
Alright that's an interesting point, although I think it's bit of overkill worrying about branch predictions hints in this method, as I don't see this being called more than once or twice in a project.

Reply:
Yeah it was more about the second point that the debugger will show users that the error is `Condition 'ssl & Object::cast_to<StreamPeerSSL>(p_connection.ptr() == nullptr' is true.`, which gives a hint as to why the error complains about StreamPeerSSL specifically.

But yeah it's a bit overkill.